src/channels/discord.ts

```diff
@@ -1,11 +1,49 @@
-import axios from 'axios';
+import { deferredPromise } from '@chainflip/utils/async';
+import { Client, GatewayIntentBits, type TextChannel } from 'discord.js';
+import { handleExit } from '../utils/functions.js';
+import logger from '../utils/logger.js';
 
-export type DiscordConfig = { webhookUrl: string };
+export const client = new Client({
+  intents: [GatewayIntentBits.Guilds],
+});
 
-export const sendMessage = async ({ webhookUrl }: DiscordConfig, content: string) => {
-  const response = await axios.post(webhookUrl, { content });
+export const login = async (token: string) => {
+  const { promise, resolve } = deferredPromise<undefined>();
```

Comment:
should be able to call just `resolve()` like this, don't care enough to block the ticket tho
```suggestion
  const { promise, resolve } = deferredPromise<void>();
```